### PR TITLE
Set Karpenter Pod to high priority and guaranteed

### PR DESCRIPTION
**Pull request by njtran**
Issue #, if available:
#381 #380 

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: config/controller.yaml [hunk 1]
```diff
@@ -65,6 +74,9 @@ spec:
           requests:
             cpu: 100m
             memory: 20Mi
```
Comment (ellistarn) [inline config/controller.yaml:76]: This seems way to low. Can we default to 1 cpu and 1gb?

Comment (njtran) [reply config/controller.yaml:76]: Sure! I think these are pretty low as well, but didn't know if you had these for a reason. I want this to be able to fit on nodes with 1vCPU as well (assuming not that many daemonsets). WDYT about `800m` and `1Gi` ?

Comment (ellistarn) [reply config/controller.yaml:76]: Nodes with 1vcpu (the t3.micros) have much more overhead than that 800 won't fit. Feel free to test it out ;)

Comment (njtran) [reply config/controller.yaml:76]: For reference, `kube-proxy` requests `100m` and `aws-node` requests `10m`

Comment (prateekgogia) [reply config/controller.yaml:76]: When I was doing some testing for the demo, I saw the Karpenter pod got OOM killed. IIRC, it was set to 20Mi limit. I think the real number can only be decided with scale testing. 
Can we skip setting limits for now, or do we need for something?

Comment (njtran) [reply config/controller.yaml:76]: I'd like to at least increase it to make it a little more resistant to OOM killing until we get the scale testing done.

Comment (ellistarn) [reply config/controller.yaml:76]: Agree with Nick.

**Review by ellistarn — APPROVED** [review]

**Review by ellistarn — APPROVED** [review]

**Pull request merged**